Azure/sonic-utilities
# [sfputil] Refactor to utilize sonic-platform package

#### What I did

Refactor sfputil to use sonic-platform package in lieu of old, deprecated platform plugins.

The sfputil utility is still useful, as sfpshow only reads and displays SFP data from State DB. However, we still need a utility to write to the transceivers (low power mode, reset, etc.), and the ability to read directly from the transceivers is helpful for debugging.

#### How I did it

- Complete refactor to use sonic-platform package
- Refactor display functions to align with sfpshow utility
- Bump version to 3.0
- Add unit tests

#### How to verify it

Run all `sfputil` commands, ensure they work. Do the same on multiple platforms. Output should be the same as before.

#### Previous command output (if the output of a command-line utility has changed)

#### New command output (if the output of a command-line utility has changed)
